### PR TITLE
Multi-step log in/sign up modal, with user type

### PR DESCRIPTION
**Pull request by austensen**
- adds ability for account sign up to happen all within a modal from the building page
- adds a required step for account sign up, asking user type (multiple-choice or write in)
- adds verify email/ re-send email step to the end of log in/sign up flow
- refactors email input into separate component (like `PasswordInput`)
  - email, password, and user type input components now work with custom `useInput` hook so that the parent can keep track of values in state. 
  - This makes it easier for the parent component to hold onto input values from each step to use for error alerts and registration at the end. The same setup also helps with the input-level errors for email and whether they shown. 
  - This still feels a bit overcomplicated, but we'll figure this all our for the component library and hopefully swap those in soon. 

https://github.com/JustFixNYC/auth-provider/pull/14 adds this new field to the user model, and https://github.com/JustFixNYC/auth-provider/pull/18 adds it to the `user/register` auth endpoint

[sc-13535]
[sc-13510]
[sc-13594]
[sc-13532]

### CODE REVIEW

File: client/src/components/AuthClient.ts
Comment (kiwansim) [inline client/src/components/AuthClient.ts:50]: nit, but I think our convention for JS variables is camel case, so `userType`

File: client/src/components/EmailAlertSignup.tsx
Comment (kiwansim) [inline client/src/components/EmailAlertSignup.tsx:56]: `{email}` is showing up as plaintext here!
Screenshot 2024-03-05 at 2 27 49 PM

File: client/src/components/Login.tsx
Comment (kiwansim) [inline client/src/components/Login.tsx:22]: this is a much clearer way of defining the login states, thank u!!

File: client/src/components/Login.tsx
Comment (kiwansim) [inline client/src/components/Login.tsx:52]: should this be set to the `registerInModal` prop value?

File: client/src/components/Login.tsx
Comment (kiwansim) [inline client/src/components/Login.tsx:439]: remove commented out bits. surprised the linter missed this!

File: jfauthprovider/views.py
Comment (kiwansim) [inline jfauthprovider/views.py:80]: referring back to the first comment, the python side of this can be 
`"user_type": request.POST.get("userType")`

File: client/src/util/helpers.ts
Comment (kiwansim) [inline client/src/util/helpers.ts:135]: this is a really neat new pattern

**Review by kiwansim — APPROVED** [review]
wow massive props to you! I'm going to approve so I don't block the merge and my comments are small nits!

File: client/src/components/Login.tsx
Comment (austensen) [reply client/src/components/Login.tsx:52]: no, sorry it's a bit confusing. `registerInModal` is for whether it can ever be a modal (ie on the building page, rather than nav login). and `showRegisterModal` controls if it's actually open, and we only want that if they submit their email on the building page and it's not associated with a user so that it opens the modal for registration rather than staying on page for login.

File: jfauthprovider/views.py
Comment (austensen) [reply jfauthprovider/views.py:80]: Thanks for catching this. After looking back in AuthClient I realized all the other methods made the change there so I followed that for consistency

File: client/src/components/EmailAlertSignup.tsx
Comment (austensen) [reply client/src/components/EmailAlertSignup.tsx:56]: It's working on local so I'm pretty sure this is just that issue with not having done `lingui extract` yet, but I've been holding off until everything is merged into `email-alerts..`

**Pull request merged**
